---
title: "Exploring population structure with admixture models and principal components analysis"
author: "Chi-Chun Liu, Suyash Shringapure, David Alexander, Ken Lange, John Novembre"
date:
output:
  html_document:
    fig_caption: yes
  pdf_document:
    fig_caption: yes
csl: springer-basic-author-date.csl.txt
bibliography: references.bib
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(fig.width=6, fig.height=4, fig.path='plot/',
                      echo=TRUE, eval=FALSE, warning=TRUE, message=TRUE)
```



Population structure is a commonplace feature of genetic variation data, and it has importance in numerous application areas, including evolutionary genetics, conservation genetics, and human genetics.  At a broad level, population structure is the existence of differing levels of genetic relatedness among some subgroups within a sample.  This may be due to the existence of close relatives in a dataset (more precisely called family structure), though most often it because samples have been drawn from geographically distinct sub-populations or different locales across a geographic continuum.  Regardless, understanding the structure in a sample is necessary before more sophisticated analyses are undertaken. For example, to infer divergence times between two populations requires knowing the existence of those two populations and which individuals belong to each.  

Two of the most commonly used approaches to describe population structure in a sample are using principal components analysis [@Menozzi78, @CavSforza94, @Price06, @Patterson06] and the Pritchard-Stephens-Donnelly model of admixture [@Pritchard00, @Novembre16a].  In brief, principal components analysis reduces a highly multi-dimensional dataset to a much smaller number of dimensions that allows for visual exploration and compact quantitive summaries.  In the application to genetic data, the multiple dimensions are the numerous genotypes observed per individual; such that it is not uncommon for a genetic dataset to have hundreds of thousands of observed dimensions per individual. The Pritchard-Stephens-Donnelly model of admixture is a model in which individuals in a sample are envisioned as having a proportion of their genome derived from each of several source populations.  Methods using this model return the proportions of ancestry in each source populations, and these make for nice compact visual summaries that reveal the existence of population structure in a sample.  

The history and basic behaviors of both these approaches have been written about extentsively, including by some of us, and so we refer readers to those previous publications to learn the basic background and interpretative nuances of these approaches and their derivatives [@Pritchard00, @Falush03, @Rosenberg05, @Hubisz09, @Raj14, @Novembre14, @Falush16, @Alexander09, @Alexander11, @Price06, @Patterson06, @Novembre08a, @McVean09, @Novembre16b].Here, we provide a protocol for running these analyses and share some pragmatic caveats that do not always arise in more abstract discussions regarding these methods.  

## Materials 
The protocol we present is based on two pieces of software: 1)  the `ADMIXTURE` software that our team developed [@Alexander09] for efficiently estimating admixture proportions in the PSD model.  2) The `smartpca` software developed by Nick Patterson and colleagues for carrying out PCA [@Price06] (?).  Both of these pieces of software are used widely.  We also pair them with downstream tools for visualization, in particular `pong` [@Behr16], for visualizing output of admixture proportion inferences, and `PCAviz` [@Novembre17] , a novel R package for plotting PCA outputs.  We also use `PLINK` [@Purcell07, @Chang15] as a tool to perform some basic manipulations of the data. 

The example data we use is derived from publicly available single-nucleotide polymorphism (SNP) genotype data from the Human Genome Diversity Panel [@Cann02].  Specifically, we will look at Illumina 650Y genotyping array data from the CEPH-Human Genome Diversity Panel, as first described by Li et al [@Li08].  This sample is a global-scale sampling of human diversity with 52 populations in total, and the raw files are available from the
following link:
<http://hagsc.org/hgdp/files.html>.  These data have been used in numerous subsequent publications and are an important reference set.  

A few technical details are that the genotypes were filtered with an Illumina GenCall score [@GenCall] cutoff of 0.25 (a quality score generated by the basic genotype calling software).  Individuals with a genotype call rate <98.5% were removed, with the logic being that if a sample has many missing genotypes it may due to poor quality of the source DNA, and so none of the genotypes should be trusted.  Beyond this, to prepare the data, we have filtered down the individuals to a set of 938 unrelated individuals.  (The starting data are available as plink-formatted files `H938.bed` 
`H938.fam`, `H938.bim`, and an accompanying set of population identifiers `H938.clst.txt`, from this link: <https://www.dropbox.com/sh/66p1hhbtx0rb3wt/AAA0aDXVTNhjUi5nH03__D4za?dl=0>).   

As a pragmatic side note, it is common (and recommended) when carrying out analyses of population structure to merge one's data with other datasets that contain populations which may be representative sources of admixing individuals. For example, in analyzing a dataset with African American individuals, it can be helpful to include datasets containing African and European individuals in the analysis. These datasets can be merged with your dataset using software such as `plink`. However, when merging several datasets, one should be aware of potential biases that can be introduced due to strand flips (i.e. one dataset reports genotypes on the '+' strand of the reference human genome, and another on the '-' strand). One precautionary step to detect strand flips is to group individuals by what dataset they derive from and then produce a scatterplot of allele frequencies for pairs of groups at a time.  If strand flips are not being controlled correctly, one will observe numerous variants on the $y=1-x$ line, where $x$ is the frequency in one dataset and $y$ is the frequency in a second dataset.  (Note: this rule of thumb assumes levels of differentiation are low between datasets, as is the case in human datasets in general). 

## Methods

In this section we walk you through an example analysis using `ADMIXTURE` and `smartpca`.  We assume the raw data files are in a directory 'raw_input' that is a sister directory to our working directory which we will call 'analysis'.  

### Subsetting Data
For running some simple examples below, we will first create a subset of the HGDP sample that is restricted to only European populations. The European populations in the HGDP have the labels 'Adygei', 'Basque', 'French', 'Italian', 'Orcadian', and 'Sardinian', so we create a list of individuals matching these labels using an `awk` command, and then use `plink`'s `--keep` option to make a new dataset with output prefix 'H938\_Euro'. 

```{r, engine = 'bash', eval = FALSE}
awk '$3=="Adygei"||$3=="French_Basque"||$3=="French"|| \
$3=="North_Italian"||$3=="Orcadian"||$3=="Russian"|| \ 
$3=="Sardinian"||$3=="Tuscan" {print $0}' raw_input/H938.clst.txt > out/Euro.clst.txt

plink --bfile raw_input/H938 --keep Euro.clst.txt --make-bed --out out/H938_Euro
```

### Filter out SNPs to remove linkage disequilibrium (LD).  
SNPs in high LD with each other contain redundant information.  More worrisome is the potential for some regions of the genome to have a disproportionate influence on the results and thus distort the representation of genome-wide structure. A standard approach to address this issue is to filter out SNPs based on pairwise LD to a reduced set of more independent markers.  Here we use `plink`'s commands to produce a new LD-pruned dataset with output prefix 'H938.LDpruned'.  The approach considers a chromosomal window of 50 SNPs at a time, and for any pair whose genotypes have an association $r^2$ value greater than 0.1, it removes a SNP from the pair.  Then the window is shifted by 10 SNPs and the procedure is repeated:  

```{r, engine = 'bash', eval = FALSE}
plink --bfile raw_input/H938 --indep-pairwise 50 10 0.1
plink --bfile raw_input/H938 --extract plink.prune.in --make-bed --out out/H938.LDprune

plink --bfile out/H938_Euro --indep-pairwise 50 10 0.1
plink --bfile out/H938_Euro --extract plink.prune.in --make-bed --out out/H938_Euro.LDprune
```

[Advanced note: For particularly sensitive results, we recommend additional rounds of SNP filtering based on observed principal component loadings and/or population differentiation statistics. In particular, one wants to assess that the results represent genome-wide structure rather than the influence of one or a few genomic regions.  See discussion below].
    
### Running `ADMIXTURE`
The `ADMIXTURE` software (v 1.3.0 here) comes as a pre-compiled binary executable file for either Linux or Mac operating systems. To install, simply download the package and move the executable into your standard execution path (e.g. '/usr/local/bin' on many linux systems).  Once installed, it is straightforward to run `ADMIXTURE` for a fixed setting for the number of source populations, commonly denoted by $K$. For example, to get started let's run ADMIXTURE with $K$=8 as there are nominally 8 populations in the data:
```{r, engine = 'bash', eval = FALSE}
admixture out/H938_Euro.LDprune.bed 8
```    
`ADMIXTURE` is a maximum-likelihood based method, so as the method runs, you will see updates to the log-likelihood as it converges on a solution for the ancestry proportions and allele frequencies that maximize the likelhood function.  The algorithm will stop when the difference between successive iterations is small (the 'delta' value takes a small value).  A final output is an estimated $F_{ST}$ [@Kent09] (?) between each of the source populations, based on the inferred allele frequencies.  These estimates reflect how differentiated the source populations are, which is important for understanding whether the population structure observed in a sample is substantial or not (values closer to 0 reflect less population differentiation).

After running, `ADMIXTURE` produces two major output files.  The file with suffix `.P` contains an $L \times K$ table of the allele frequencies inferred for each SNP in each population.  The file with suffix `.Q` contains an $N \times K$ table of inferred individual ancestry proportions from the $K$ ancestral populations, with one row per individual. 

For our example dataset with $K$=8, this will be a file called `H938.LDpruned.8.Q`. This file can be used to generate a plot showin individual ancestry. In `R`, this can be done using the following commands:
```{r, eval = FALSE,fig.height=2.5,fig.width=7.5}
library(RColorBrewer)
tbl=read.table("out/H938_Euro.LDprune.8.Q")
par(mar=c(1,4,2,2))
barplot(t(as.matrix(tbl)), col=brewer.pal(8,"Set1"),
        ylab="Admixture Proportions",border=NA,space=0)
```
Each thin vertical line in the barplot represents one individual and each color represents one inferred ancestral population. The length of each color in a vertical bar represents the proportion of that individual's ancestry that is derived from the inferred ancestral population corresponding to that color. The above image suggests there are some genetic clusters in the data, but it's not a well organized data display. 

To improve the visualization, one can use a package dedicated to plotting ancestry proportions [@Rosenberg04, @Kopelman15, @Behr16]. Here we use a post-processing tool `pong` [@Behr16], which visualizes individual ancestry with similarity between individuals within clusters. You will most likely want to install 'pong' on a local machine as it initializes a local web server to display the results.  

To run `pong` requires setting up a few files: 1) an `ind2pop` file that maps individuals to populations 2) a `Qfilemap` file that points `pong` towards which '.Q' files to display. These are easy to build up from the command-line based using the `Euro.clst.txt` file we built above, and an 'awk' command to output tab-seperated text to a file with the `Qfilemap` suffix added to whatever file prefix we're using to organize our runs:

```{r, engine='bash',eval=FALSE}
cut -d' ' -f3 out/Euro.clst.txt > out/H938_Euro.ind2pop
FILEPREFIX=out/H938_Euro.LDPrune
K=8
awk -v K=$K -v file=$FILEPREFIX 'BEGIN{ \
printf("ExampleRun\t%d\t%s.%d.Q\n",K,file,K) 
}' > $FILEPREFIX.Qfilemap 
```
Note when building the '.Qfilemap' one needs to use tabs to seperate the columns for 'pong' to read the file correctly.

Then to run 'pong', we use following command:
```{r, engine = 'bash', eval = FALSE}
pong -m out/H938_Euro.LDprune.Qfilemap -i out/H938_Euro.ind2pop
```       
and open a web browser to 'http://localhost:4000/' to view the results. From this visualization we can see the admixture model fits Adygei, Orcadian, Russian as largely being derived each from a single source population (represented by orange, green, and blue respectively); the Sardinian and the French Basque samples are modeled as comprising individuals from two source populations each (yellow/red for Sardinia, brown/pink for French Basque); and the French, Tuscan, and North Italian samples are generally estimated to have a majority component of ancestry from a single source population ('purple') though with notable admixture with other sources.

![`pong` visualization of `admixture` output \label{figure.1}](./plot/H938_Euro.LDprune.K8.png)


### Considering different values of $K$
In a typical analysis, one typically wants to explore the sensitivity of the results to the choice of $K$.  One approach is to run ADMIXTURE with various plausible values of $K$ and compare the performance of results visually and using cross-validation performance. Here is a piece of `bash` command-line code that will run `ADMIXTURE` for values of $K$ from 2 to 12, and that will build a file with a table of cross-validation error rates per value of $K$.  

```{r, engine = 'bash', eval = FALSE}
# Run for different values of K
prefix=H938_Euro.LDprune
Klow=1
Khigh=12
for ((K=$Klow;K<=$Khigh;K++)); \
do 
  admixture --cv out/$prefix.bed $K | tee log.$prefix.${K}.out; 
done
```
Then let's compile results on cross-validation error across values of K:
```{r, engine = 'bash', eval = FALSE}
echo '# CV results' > $prefix.CV.txt
for ((K=$Klow;K<=$Khigh;K++)); do 
  awk -v K=$K '
    $1=="CV"{print K,$4}' out/log.$prefix.$K.out >> out/$prefix.CV.txt; 
done
```
and build a Qmapfile for pong:
```{r, engine='bash', eval = FALSE}
echo '# Qfilemap' > out/$prefix.Qfilemap
for ((K=$Klow;K<=$Khigh;K++)); \
do 
  awk -v K=$K -v prefix=out/$prefix 'BEGIN{ \
    printf("K%d\t%d\t%s.%d.Q\n",K,K,prefix,K)
  }' >> out/$prefix.Qfilemap; 
done
```     

Now let's inspect the outputs.  First let's make a plot of the cross-validation error as a function of $K$:
```{r,eval = FALSE,fig.width=3.25,fig.height=3}
tbl = read.table("out/H938_Euro.LDprune.CV.txt")
plot(tbl$V1,tbl$V2,xlab="K",ylab="Cross-validation error",pch=16,type='l')
```
The cross-validation error suggests a single source population can model the data adequately and larger values of $K$ lead to over-fitting.  Let's now inspect the 'pong' outputs. 

```{r, engine = 'bash', eval = FALSE}
pong -m out/H938_Euro.LDprune.Qfilemap -i out/H938_Euro.ind2pop
```  

[TODO insert png of output and describe]

### Some advanced options

Running `ADMIXTURE` with the `-B` option provides estimates of standard errors on the ancestry proportion inferences. The `-l` flag runs `ADMIXTURE` with a penalized likelihood that favors more sparse solutions (i.e.~ancestry proportions that are closer to zero).  This is useful in settings where small, possibly erroneous ancestry proportions may be overinterpreted. For two datasets with the same set of SNPs, the cluster-level allele frequencies inferred from one dataset can be provided as input for inference of admixture proportions in a second dataset, by using the `-P` option. This is useful individuals of unknown ancestry are being analyzed against the background of a reference sample set.  Please see the `ADMIXTURE` manual for a complete listing of options though. 
    
### PCA with SMARTPCA
Comparing ADMIXTURE and PCA results often helps give insight and confirmation regarding population structure ina  sample. To run PCA, a standard package that is well-suited for SNP data is the `smartpca` package maintained by Nick Patterson and Alkes Price at (http://data.broadinstitute.org/alkesgroup/EIGENSOFT/). To run it, we first set-up a basic smartpca parameter file from the command-line:

```{r, engine = 'bash', eval = FALSE}
PREFIX=H938_Euro.LDprune
echo genotypename: out/$PREFIX.bed > out/$PREFIX.par
echo snpname: out/H938_Euro.LDprune.bim >> out/$PREFIX.par
echo indivname: out/H938_Euro.LDprune.PCA.fam >> out/$PREFIX.par
echo snpweightoutname: out/H938_Euro.LDprune.snpeigs >> out/$PREFIX.par
echo evecoutname: out/H938_Euro.LDprune.eigs >> out/$PREFIX.par
echo evaloutname: out/H938_Euro.LDprune.eval >> out/$PREFIX.par
echo phylipoutname: out/H938_Euro.LDprune.fst >> out/$PREFIX.par
echo numoutevec: 20 >> out/$PREFIX.par
echo numoutlieriter: 0 >> out/$PREFIX.par
echo outlieroutname: out/H938_Euro.LDprune.out >> out/$PREFIX.par
echo altnormstyle: NO >> out/$PREFIX.par
echo missingmode: NO >> out/$PREFIX.par
echo nsnpldregress: 0 >> out/$PREFIX.par
echo noxdata: YES >> out/$PREFIX.par
echo nomalexhet: YES >> out/$PREFIX.par
```   
This input parameter file runs `smartpca` in its most basic mode (i.e. no automatic outlier removal or adjustments for LD - features which you might want to explore later). Note: You may need to change `../data/` to reflect the real path where your files are.

As a minor issue, `smartpca` ignores individuals in the .fam file if they are marked as missing in the phenotypes column. This `awk` command replaces the pheonotype column column with a column of one. You may skip this command if your data is in the eigenstrat format.
    
```{r, engine = 'bash', eval = FALSE}
awk '{print $1,$2,$3,$4,$5,1}' out/H938_Euro.LDprune.fam > out/H938_Euro.LDprune.PCA.fam
```
Now run smartpca with the following command. You may find the output files at the paths specified in the parameter file.
    
```{r, engine = 'bash', eval = FALSE}
smartpca -p ./out/H938_Euro.LDprune.par
```   

#### Plotting PCA results with PCAviz
The PCAviz package can be found at (https://github.com/NovembreLab/PCAviz). It provides a simple interface for quickly creating visually compelling plots from Principal Components Analysis (PCA) and accompanying data. It encodes several of the best practices we use for plotting PCA (such as using abberviations for point characters and plotting median positions of each labelled group).  The installation is simple:
```{r, eval=FALSE,echo=FALSE}
install.packages("devtools")
devtools::install_github("NovembreLab/PCAviz",build_vignettes = TRUE)
```

The following command in `R` generates plots showing each individual sample's position in the PCA space and the median position of each labelled group in PCA space:

```{r, eval = FALSE,echo=FALSE, results='hide',warning=FALSE,message=FALSE, fig.height=10,fig.width=10}
library(PCAviz)
library(cowplot)
prefix="out/H938_Euro.LDprune"
nPCs = 20

#read in individual coordinates on PCs and eignvalues
PCA=read.table(paste(prefix,".eigs",sep=""))
names(PCA)=c("ID",paste("PC",(1:nPCs),sep=""),"case.control")
PCA=PCA[,1:(nPCs+1)] # Remove case/control column
eig.val = sqrt(unlist(read.table(paste(prefix,".eval",sep="")))[1:nPCs])
sum.eig= sum(unlist(read.table(paste(prefix,".eval",sep=""))))

# Read in snp weightings matrix
snpeigs = read.table(paste(prefix,".snpeigs",sep=""))
names(snpeigs)=c("ID","chr","pos",paste("PC",(1:nPCs),sep=""))
snpeigs$chr = factor(snpeigs$chr)
rownames(snpeigs) <- snpeigs$ID; snpeigs = snpeigs[,-1]

#Note smartpca pushes the plink family and individual ids together so we need to extract out the ids afresh
#This is a little ugly but gets the job done
tmp=unlist(sapply(as.character(PCA$ID),strsplit,":"))
ids=tmp[seq(2,length(tmp),by=2)]
PCA$ID=ids

# Read in the group/cluster labels 
clst=read.table("out/Euro.clst.txt")
clst_unord=clst$V3[match(ids,clst$V2)]  # Order them to match the ids of PCA object
PCA = as.data.frame(PCA)
PCA = cbind(PCA,clst_unord)
names(PCA)[ncol(PCA)] <- "sample"

# Build the PCAviz object
hgdp <- pcaviz(dat = PCA, sdev=eig.val, var= sum.eig, rotation = snpeigs)
hgdp <- pcaviz_abbreviate_var(hgdp,"sample")

# Make PCAviz plots by choosing coords
geom.point.summary.params = list(shape = 16,stroke = 1,size = 5,
                                 alpha = .7,show.legend = F)
plot1 <- plot(hgdp,coords = paste0("PC",c(1,2)), color = "sample",
              geom.point.summary.params = geom.point.summary.params ,scale.pc.axes = 0.6)
plot2 <- plot(hgdp,coords = paste0("PC",c(2,3)), color = "sample",
              geom.point.summary.params = geom.point.summary.params ,scale.pc.axes = 0.6)
plot3 <- plot(hgdp,coords = paste0("PC",c(4,5)), color = "sample",
              geom.point.summary.params = geom.point.summary.params ,scale.pc.axes = 0.6)
plot4 <- plot(hgdp,coords = paste0("PC",c(5,6)), color = "sample",
              geom.point.summary.params = geom.point.summary.params ,scale.pc.axes = 0.6)
plot_grid(plot1,plot2,plot3,plot4)
```

First one may notice population isolates are again separated with PC1 and PC2, with Orcadian projected close to France, similar to the result in `ADMIXTURE`. PC3 further distinguishes Orcadian to the others. PC7 corresponds to two clustering observed within Sardinians in the `ADMIXTURE` analysis. Lower PCs also pick up individual difference within population, especially within Adygei (See PC6 vs. PC7 for example). It may be due to variance in sample qualities or cryptic structures in Adygei.

```{r, eval = FALSE,fig.height=15,fig.width=10}
pcaviz_violin(hgdp,pc.dims =paste0("PC", c(1:10)),plot.grid.params = list(nrow = 5))
```

Information for each SNP is also useful, in particular loading is measuring the correlation between a SNP and a PC. Thus if PCs are caputuring structures in the data, outlier snps in terms of loading may indicate selection. [@Duforet-Frebourg16]

```{r, eval = FALSE,fig.height=30,fig.width=8}
for (i in 1:10){
  plotname<-paste("plot",i,sep="")
  plot <- pcaviz_loadingsplot(hgdp,pc.dim=paste0("PC",i), 
                              min.rank=0.8,gap=200,color = 'chr') +
          xlab("SNPs") + ylab(paste0("PC",i," loading"))
  assign(plotname,plot)
}

plot_grid(plot1,plot2,plot3,plot4,plot5,plot6,plot7,plot8,plot9,plot10, nrow=10, align="v")

```

The proportion of total variance explained by each PC is a useful set of quantities for understanding structure ina a sample and for evaluating how many PCs one might want to include in downstream analyses. This can be computed as  $\lambda_i / \sum_{k} \lambda_k$, with $\lambda_i$ being eigenvalues in decreasing order, and is plotted below.
```{r, eval = FALSE,warnings=FALSE,fig.width=3.25,fig.height=3}
screeplot(hgdp,type='pve') + ylim(0.005,.018) +
      theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
      theme(axis.line = element_line(size = 1, linetype = "solid"))
```

##  Discussion

Our protocol above is relatively straightfoward and presents the most basic implementation of these analyses. Each analysis sofware (`ADMIXTURE` and `smartpca`) and each visualization package ('pong' and 'PCAviz') contains numerous other options that may be suitable for specific analyses and we encourage the readers to spend time in the manuals of each.  Nonetheless, what we have presented is a useful start and a standard pipeline that we use in our research.

Several of the prominent reviews on the performance of these methods will highlight the confusing impacts of factors such as uneven sample sizes across sub-populations, the inclusion of close relatives in a sample (cryptic relatedness), spatial structure, or small sample sizes.  We do not want to belabor these points, but we do want to highlight the impact of localized genomic regions that have extensive marker-to-marker association (linkage disequilibrium) on PCA and admixture model results.  The issue is mainly discussed amongst in person among experienced analysts without having a concomitantly large discussion in the published literture. This issue is particularly prominent when working with human genetic data with low levels of differentiation, as in those settings, the methods seem more prone to fitting structure in particular genomic regions versus a weak single of structure across the whole genome.  A nice empirical example of the problem is in XXXX, where PCX is seen to be indicative of an inversion type at the 8pXXX region.  [also Cite Peter Ralph bioRxiv?]

Two broad perspecives we find helpful are to always keep in mind 1) How the admixture model and PCA framework are related to each other indirectly as different forms of sparse factor analyss [@Engelhardt10]; 2) How the PCA framework in particular can be considered as a form of efficient data compression.  Both of these perspectives can be helpful in interpreting the outputs of the methods and for appreciating how these approaches best serve as helpful visual exploratory tools for analyzing structure in genetic data.  The methods produce characterizations of structure using simplified statistical tools for complex species, and though the results rarely resolve specific hypotheses directly, they are extremely useful for framing specific models of population structure that can be further investigated using more detailed and explicit approachs; for example, those based on coalescent or diffusion theory (Chapters XX and XX). 

## Extra-text (Can be included in discussion or removed for brevity)

First run ADMIXTURE on all 938 individuals. In figure 1, with K=5, 6 continental groups are separated. The 6-th component further separates European from Central/South Asian, and captures the uniqueness of Sardinian. The 7-th component separates middle-east populations from European. In Rosenberg et al, East Asia with little among-population variance component, requires almost all loci to achieve comparable results to that using full data (Also see 3.2 Troubleshooting). One may observe in Yakut, whose language is Altaic, is the only population in East Asia that share ancestry with America and Europe [@Rosenberg02;@Li08].

```{r}
#![ADMIXTURE with all 938 individuals \label{figure.1}](./plot/mainviz_2017-10-12_19h59m48s_pong.png)
```

Although populations in Europe seem to be homogeneous, at finer-scale with only European individuals, ADMIXTURE with K=6 separates Europe populations (figure 2). Specifically, individuals from the population isolates (Adygei, Baseque, Orcadian, and Sardian) are inferred to have distinct ancestral populations, while French and Italian share similar ancestry. And please note that Sardinians are inferred to have two ancestral populations.


```{r} 
#![ADMIXTURE with all European individuals. \label{figure.2}](./plot/K=6_majormode_2017-10-12_20h7m0s_pong.png)
```


ADMIXTURE analyses may at times produce results that may seem contradictory with known history of populations. This can be a true representation of previously unknown history, or suggestive of some common problems with the analysis. Some factors that can cause problems with ADMIXTURE analyses are:

1. Imbalanced sample sizes (two different views) : ADMIXTURE assumes that the sample sizes from various populations in the analysis are roughly representative of their sample sizes in the real world. Therefore, if the dataset to be analyzed is severely imbalanced and contains too many or too few individuals from a population, that may produce misleading ADMIXTURE results [@Puechmaille16, @Shringarpure14]. Some prior knowledge about populations is therefore essential before ADMIXTURE analysis. Recent investigation points out a population with relatively few samples tends to end up as mixes of several groups, which might be the case for combining ancient and modern samples [@Falush16]. In the same artical, authors use analysis in Friedlaender 2008 as an example, in which oversampling Melanesians gives different ADMIXTURE results. A sampling scheme proportional to modern population sizes does not imply a more correct result, and it’s hard to define a best sampling scheme.

2. Too few SNPs : ADMIXTURE needs more markers to differentiate populations. As a rule of thumb, differentiating between populations with $Fst \approx 0.1$ requires 10,000 SNPs, and this number increases inversely with decreasing Fst. Readers can refer to [@Alexander09] for detail investigations. But coupled with modeling assumtion, this issue becomes more intricate.

3. Modeling assumption (directly from papers and not organized, credit to [@Lawson12, @Patterson12]):
    +  ADMIXTURE assume all underlying populations undergo separate genetic drift from some original founder group, and so this prior model penalises shared drift, for every individual marker, and so increasingly strongly as the number of loci increases, and this assumption also makes results difficult to interpret historically.

4. How to not over-interpret results: ADMIXTURE may infer similar results under different demographic histories. with simulation, both unsampled ancestral population and bottleneck give indistinguishable individual admixture proportion to recent admixture. A misinterpretation from the bottleneck scenario actually had happened in an analysis of relationships between Ari Blacksmiths and Ari Cultivators in Ethiopia. The limitations of ADMIXTURE analyses and the pitfalls of over-interpreting ADMIXTURE results are described in more detail in [@Falush16].

5. Choosing an optimal K:  The value of K with minimum cross-validation error provides the best statistical fit. ADMIXTURE essentially tries to explain variation between individuals with a model. If there’s no variation between individuals, ADMIXTURE can attain the same likelihood simply by shifting allele frequencies. This causes the inferred K tend to be smaller than the number of distinct drifts. [@Falush16] In addition, sometimes it is possible that no value of K in ADMIXTURE can capture the true variation in the data accurately and a different set of models is required.

6. Examining model fit: The cross-validation procedure described earlier provides one way of evaluating model fit, but sometimes admixture model may poorly represent the underlying mechanism. For example, if the data being analyzed comes from a population where genetic variation lies along a cline rather than in clusters, an explicit spatial model of genetic variation may be more appropriate [@Francois06]. Recently, a few approaches have been proposed to evaluate model fit. In [@Falush16], The authors propose to paint individual chromosome using other individuals as complimentary information to ADMIXTURE. If recent admixture is the true underlying population history, we should be able to recover individual palette proportion by combining ancestral palette and individual admixture proportion. Systematic deviations from the predicted individual palette imply inconsistency with recent admixture. For Bayesian approaches (e.g. STRUCTURE), in [@Mimno15] the authors propose posterior predictive checks as measurement on misspecification of admixture models. If the recent admixture is the true model, observed data and replicated data simulated with inferred parameters should have similar properties. And discrepancies between replicated data and observed real data could be measured from statistics in population genetics such as LD, IBS and Fst, etc. Specifically with IBS, parameters inferred in POPRES data [@Nelson08] underestimate individual similarity, and these individuals are known to follow isolation-by-distance [@Novembre08b] and are thus not well separated.

It may also be useful to compare the result with the one obtained from non-model based approaches, on which corrections for spatial correlation have been developed [@Frichot12]. Using those methods can be helpful to decide if your ADMIXTURE model fit is appropriate for your dataset.            

Some notes on PCA:

1. missing data: Orthogonal projection with a lot of missing sites may be very unstable. An alternative solution is instead resort to least square projection (lsqproject option in smartpca). This is especially important for ancient DNA samples, which ususally are of lower quality and highly damaged.

2. scaling: smartpca scales the genotype matrix as unit variance. This will assign higher weights to rare variants. Since rare variants may be segregated in only in specific populations, this scaling scheme might increase the ability to detect finer structure.

PCA and ADMIXTURE can both viewed as sparse factor analysis with different constraints [@Engelhardt10]. Constraints assumed by ADMIXTURE is stronger than that by PCA, which leads to its ability of detecting well seperated population but not populations spread along continuum  


## References
